packages/eslint-plugin/docs/rules/no-unnecessary-qualifier.md

```diff
@@ -0,0 +1,58 @@
+# Warns when a namespace qualifier is unnecessary. (no-unnecessary-qualifier)
+
+## Rule Details
+
+This rule aims to let users know when a namespace or enum qualifier is unnecessary,
+whether used for a type or for a value.
+
+Examples of **incorrect** code for this rule:
+
+```ts
+namespace A {
+  export type B = number;
+  const x: A.B = 3;
+}
+```
+
+```ts
+namespace A {
+  export const x = 3;
+  export const y = A.x;
+}
+```
+
+```ts
+enum A {
+  B,
+  C = A.B
+}
+```
+
+```ts
+namespace A {
+  export namespace B {
+    export type T = number;
+    const x: A.B.T = 3;
+  }
+}
+```
+
+Examples of **correct** code for this rule:
```

Comment:
can you please add an example for correct enum usage, and one for a correct simple (non-nested) namespace?

Just to keep clarity/parity with the incorrect and correct sets.

Reply:
Added.